waspc/examples/todoApp/src/client/pages/ProfilePage.tsx

```diff
@@ -13,10 +15,36 @@ export const ProfilePage = ({
 }: {
   user: User
 }) => {
+  const [messages, setMessages] = useState<{ id: string, username: string, text: string }[]>([]);
+  const { socket, isConnected } = useSocket<ServerToClientEvents, ClientToServerEvents>()
+  const inputRef = useRef<HTMLInputElement>(null)
+
   useEffect(() => {
     fetchCustomRoute()
+
+    socket.on('chatMessage', logMessage)
+
+    return () => {
+      socket.off('chatMessage', logMessage)
```

Comment:
This is a really cool idea @Martinsos and it goes in line with some other React libs for sockets that I saw.

I'd love to do something like `const { isConnected, registerHandler } = useSocket()`

and then `registerHandler('event', (even) => {})`

We would then use `useEffect` inside and the user would avoid a common pitfall (not using useEffect keep registering the handlers on rerenders -> happened to me today)

Reply:
Ok, sounds good, I can add that then 👍🏻

Reply:
I just thought maybe that is nicer instead of writing their own useEffect that has to do on and off! But if you think it is a bit too much we don't have to do it.